include/drivers/st25r3911b_nfca.h

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef ST25R3911B_NFCA_H_
+#define ST25R3911B_NFCA_H_
+
+#include <kernel.h>
+#include <zephyr/types.h>
+
+/**
+ * @file
+ * @defgroup st25r3911b_nfca ST25R3911B NFC Reader NFC-A Technology
+ * @{
+ *
+ * @brief API for the ST25R3911B NFC Reader NFC-A Technology.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** NFC-A Events count. */
+#define NFCA_EVENT_CNT 2
+
+/** NFC-A CRC length. */
+#define NFCA_CRC_LEN 2
+
+/** NFC-A NFCID1 maximum length. */
+#define NFCA_NFCID1_MAX_LEN 10
+
+/**
+ * @defgroup st25r3911b_error NFC-A protocol and transmission error.
+ * @{
+ */
+#define NFCA_ERR_CRC 1
+#define NFCA_ERR_PARITY 2
+#define NFCA_ERR_FRAMING 3
+#define NFCA_ERR_SOFT_FRAMING 4
+#define NFCA_ERR_HARD_FRAMING 5
+#define NFCA_ERR_LAST_BYTE_INCOMPLETE 6
+#define NFCA_ERR_TRANSMISSION 7
+
+/**
+ * @}
+ */
+
+/** @brief NFC-A Tag types.
+ */
+enum nfca_tag_type {
+	NFCA_T2T,
+	NFCA_T4T,
+	NFCA_NFCDEP,
+	NFCA_T4T_NFCDEP,
+	NFCA_T1T
+};
+
+/** @brief NFC-A Tag detect command.
+ */
+enum nfca_detect_cmd {
+	DETECT_CMD_ALL_REQ,
+	DETECT_CMD_SENS_REQ
```

Comment:
You should use a consistent namespace and at least add `NFCA_` here. But actually it would be better if all the stuff defined here used a unified prefix (in lowercase or uppercase depending on the defined entity).